### DISCUSSION TITLE
Combining motion and USB power on modes

### DISCUSSION THREAD

**Discussion by kmeine** [gopro/labs#197] [Q&A]
I'm trying to get a Hero8 to boot when USB power is on- but not to just start capturing immediately like a dashcam. Instead, I'm trying to get it to *then* capture when a motion threshold is met- basically trying to do a wildlife cam thing where the camera can be put into standby remotely via USB power.

I'm working with a code like this:

!uN!SM9H2!u2!R

And it boots with USB, but then doesn't seem to want to record.

Does this need to be implemented with a WAKE command instead?

EDIT: So I wised up and realized I probably need to do some BOOT stuff. This seems to be getting close:

!uN!MBOOT="!Lcam"!SAVEcam="!SM1!R"

But it still isn't quite behaving.

**Comment by dnewman-gpsw** [comment]
You likely need  !MWAKE=2 <- wake on power.  

What is missing a conditional sleep command, as your camera will stay on.  However if you remove the battery, using USB to power and trigger, your off condition is handled.  ;)  With the WAKE enabled, the camera will boot, and go into motion detect, and stay there until you remove the power.

**Reply by kmeine** [reply]
Sure, I had thought of something like that. I was just hoping that since the !uE USB powerdown seemed to work fine for every other command in the middle (capture now, after delay, etc.) that there was a way to make it play well with other. I was worried about file issues with just cutting the USB power with no battery in if it was still trying to record, but it seems like people do with some success.

**Reply by dnewman-gpsw** [reply]
Camera file recovery is very good, and that would only be needed if there was a capture happening at the exact time as the shutdown.  The file recovery would happen automatically upon the next boot.

**Reply by kmeine** [reply]
Alright, I'll monkey around and see what happens. Thanks!

**Comment by AthenasArch** [comment]
I think I can write on this topic.

I have the gopro 11 using it as an "HDMI" monitoring device.

I use Clean HDMI and WAKE=2, to wake up on USB power alone.

But I didn't find any command to turn off the camera when it loses USB power, I wanted to turn off the camera after x time after it lost USB power. It's possible ?

**Reply by dnewman-gpsw** [reply]
Yes, you need to run a boot script. The script can test the status of USB power, and shutdown when power it is removed.

Something like this:
<r0<u0=U+1~=U0>U9!O!1N!R

In pseudo code

```
If not recording 
  If not USB powered 
     U = U + 1
  else
     U = 0
If U > 9
   Shutdown
Sleep 1 second
Repeat
```

**Reply by AthenasArch** [reply]
I was able to run this command here: ">u0"USB ON"~"USB OFF"", to check if everything is working regarding the USB.

The problem is that I'm using the gopro only in HDMI mode, and my power supply is fluctuating a lot between "USB ON" and "USB OFF". And I don't need to use the recording mode.

when I put the basic script of: "https://gopro.github.io/labs/control/usb/" without any command, the camera keeps turning off in a very fast time interval.

Could you tell me what I can try to avoid this problem?

Could it be my power supply?

**Reply by dnewman-gpsw** [reply]
Sounds like a cable or power source problem as the >u0 test is reliable AFAIK.

**Reply by dnewman-gpsw** [reply]
The record test was a just-in-case, so the script won't shutdown mid capture

**Reply by AthenasArch** [reply]
This command seems to work "<u0=U+1!1N~U0>U9"shutdown"!R"", it checks the entire time it's connected as I seem to be having power surges of less than 5 seconds, it resets the counter.

How can I make this command permanent? that even turning off the camera will work?

**Reply by dnewman-gpsw** [reply]
This command is slightly malformed, corrected here:  <u0=U+1+!1N~=U0>U9!O!R

to make this is BOOT script

!MBOOT="!Lboot"  <- this is the command to run at boot
!SAVEboot=<u0=U+1+!1N~=U0>U9!O!R    <- this saves your script into a file MISC/boot

**Reply by dnewman-gpsw** [reply]
Sneak peak on the new Labs coding interface
![image](https://github.com/gopro/labs/assets/10856084/2c45a2fa-f559-4555-887a-b51641213d77)

Generating the above script 
![image](https://github.com/gopro/labs/assets/10856084/55787864-0e5a-45bd-baa9-761be677a65f)

Try it at https://gopro.github.io/labs/build/  (it likely has a few bugs, and it not complete.)

**Reply by AthenasArch** [reply]
I understood how the boot command works, I was able to run it on my gopro 11 and it seems to work fine. after 9 seconds without load it turns off.
If I plug in the usb cable it turns on automatically.

I'm going to test this tool, it looks amazing and it's more self-taught.

Thanks a lot for your support. I will test the [tool](https://gopro.github.io/labs/build/) and bring feedback.

**Reply by dnewman-gpsw** [reply]
The blockly tools, now adds boot command creation: 
![image](https://github.com/gopro/labs/assets/10856084/98a56ee7-bb8b-483c-991d-47a57539128d)

**Reply by AthenasArch** [reply]
Hello friend, I need to use multiple GoPro 11 cameras with the same LAB configuration.

## All settings:

- TUSB = 1
- HDMI = 1
- AGREEMENT = 2
- GPS = OFF
- WIFI = OFF
- VOICE CTRL = OFF
- QUICK CAP = OFF
- CAMERA MODE = VIDEO
-RES = 4K
- FPS = 30
- FRONT DISPLAY = DISABLE
+ This code: !MBOOT="!Lboot" + !SAVEboot=<u0=U+1+!1N~=U0>U15!O!R

I wonder if it is possible to write a single QR code to do all these updates for the other GoPro 11 cameras I will be using.

If only 1 QR Code with all the configuration is not possible, would it be possible to copy the information between SD cards already configured with this data?

What's the best way to do this on multiple cameras?

Remembering that I will have 6 cameras that must be configured the same.

Sorry for my English.

**Reply by AthenasArch** [reply]
would the command look like this?
"!MHDMI=1!MTUSB=1!MWAKE=2!SAVEboot=<u0=U+1+!1N~=U0>U15!O!R"

Another additional thing, about the QR reader. How can I guarantee that my neighbor won't put a QR Code to misconfigure my metadata?
how to deactivate the QR Code and how to reactivate it?

**Reply by AthenasArch** [reply]
Or something like this ?
![image](https://github.com/gopro/labs/assets/79209419/64f9fa30-7714-4528-9fc4-f38aac4983a8)

**Reply by dnewman-gpsw** [reply]
Not quite.  You should not do permanent metadata setups within a boot command.  You technically can, but don't do that. 

As the boot runs every time, there is no need to set permanent settings again and again.

A single QR is possible, just not with the new block tool.

You missed the command command for your single QR:
!MHDMI=1!MTUSB=1!MWAKE=2!MBOOT="!Lboot"!SAVEboot=<u0=U+1+!1N~=U0>U15!O!R
 

From the blocks
HDMI=1 is not a command, only oMHDMI=1 or !MHDMI=1 is.

> How can I guarantee that my neighbor won't put a QR Code to misconfigure my metadata?

You don't along with your neighbor?   QR Code have to be very close to be read, there is not accidental scanning.
